### ISSUE TITLE
`_endpoints` methods must not rely on `planned_units`

### ISSUE THREAD

**Issue by Abuelodelanada** [canonical/loki-k8s-operator#117]
### Bug Description

The [following method](https://github.com/canonical/loki-k8s-operator/blob/main/lib/charms/loki_k8s/v0/loki_push_api.py#L1263): 

```python
    def _endpoints(self) -> List[dict]:
        """Return a list of Loki Push Api endpoints."""
        return [{"url": self._url(unit_number=i)} for i in range(self._charm.app.planned_units())]
```

in `LokiPushApiProvider` class relies on planned units numbers to generate Loki endpoints.

But lifecycle reasons mean that at any given point unit numbers may not start with 0, may also have gaps, for instance if unit 5 is faulty I would remove it and add a new one, I'd end up with units 4, 6, 7.

### To Reproduce

There is no need to have steps to reproduce this.

### Environment

- juju 2.9.25

### Relevant log output

```shell
None
```

**Comment by Abuelodelanada** [comment]
@mmanciop said:

> that method should be easy to change based on the remote unit name, but not if it is a CMR
> Because the unit name would not be correct
> We will have to have the requesters put their topology (or at least, unit name) in their own data bags

**Comment by Abuelodelanada** [comment]
Fixed in #120
